Main.hs

```diff
@@ -10,29 +10,41 @@ import Control.Monad (unless)
 import qualified Data.List as L
 import Text.Printf (printf)
 
+aboba (a, b) = a + b
+
 multByIndex :: [Int] -> [Int]
-multByIndex = undefined
+multByIndex ls = zipWith (*) ls [0..]
 
 powerByIndex :: [Int] -> [Int]
-powerByIndex = undefined
+powerByIndex ls = zipWith (^) ls [0..]
 
 productOfDifference :: [Int] -> Int
-productOfDifference = undefined
+productOfDifference [] = error "list is too short"
+productOfDifference [_] = error "list is too short"
+productOfDifference ls = foldl (*) 1 (zipWith (-) (init ls) (tail ls))
```

Comment:
you could have done without `init ls` since zipWith produces list with the length equal to the minimum length of both lists